invenio_app_ils/mail/views.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2021 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio App ILS email views."""
+
+from flask import Blueprint
+from webargs import fields
+from webargs.flaskparser import use_kwargs
+
+from invenio_app_ils.mail.model import EmailLog
+from invenio_app_ils.permissions import need_permissions
+
+
+def serialize_email(email):
+    """Serialize email from db."""
+    if email is None:
+        return {}
+    email = email.__dict__
+    del email["_sa_instance_state"]
+    return email
+
+
+def get_emails_list_blueprint(_):
+    """Add email list view to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_emails_list", __name__)
+
+    @blueprint.route("/emails")
+    @use_kwargs(
+        {
+            "recipient_user_id": fields.Int(),
+            "loan_pid": fields.Str(),
+            "email_action": fields.Str(),
+        }
+    )
+    @need_permissions("emails-sent")
+    def get_emails(recipient_user_id=None, loan_pid=None, email_action=None):
+        filters = {}
+        if recipient_user_id:
+            filters.update({"recipient_user_id": recipient_user_id})
+        if loan_pid:
+            filters.update({"loan_pid": loan_pid})
+        if email_action:
+            filters.update({"email_action": email_action})
+
+        emails = EmailLog.query.filter_by(**filters).all()
+
+        for i, email in enumerate(emails):
+            emails[i] = serialize_email(email)
+
+        return {"hits": emails, "total": len(emails)}
+
+    return blueprint
+
+
+def get_emails_item_blueprint(_):
+    """Add emails item view to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_emails_item", __name__)
+
+    @blueprint.route("/emails/<id>")
```

Comment:
Missing param validation with `webargs`. Pay attention here because incoming values from the request should be always be sanitized.